unity-renderer/Assets/DCLPlugins/ECS7/InternalECSComponents/Interfaces/Extensions/InternalSceneBoundsCheckExtensions.cs

```diff
@@ -36,6 +35,27 @@ public static class InternalSceneBoundsCheckExtensions
             }
         }
 
+        public static void OnTransformScaleRotationChanged(this IInternalECSComponent<InternalSceneBoundsCheck> sbcInternalComponent,
+            IParcelScene scene, IDCLEntity entity)
+        {
+            var model = sbcInternalComponent.GetFor(scene, entity)?.model ?? new InternalSceneBoundsCheck();
```

Comment:
Can we avoid the allocation and use always the same instance?

Reply:
Sure, sounds good to me, done!

Reply:
there are no extra allocations on that piece of code, after a new instance is created is being stored, so next call will be using the stored instance instead of creating a new one.
that is what that line `var model = sbcInternalComponent.GetFor(scene, entity)?.model ?? new InternalSceneBoundsCheck();` does, it get the stored instance or it creates a new one if there is none stored,
2 lines later the instance get stored `sbcInternalComponent.PutFor(scene, entity, model);`